src/tests/ftest/util/file_count_test_base.py

```diff
@@ -8,84 +8,106 @@
         hdf5_plugin_path = self.params.get("plugin_path", '/run/hdf5_vol/*')
         mount_dir = self.params.get("mount_dir", "/run/dfuse/*")
         ior_np = self.params.get("np", '/run/ior/client_processes/*', 1)
         ior_ppn = self.params.get("ppn", '/run/ior/client_processes/*', None)
         mdtest_np = self.params.get("np", '/run/mdtest/client_processes/*', 1)
         mdtest_ppn = self.params.get("ppn", '/run/mdtest/client_processes/*', None)
+        intercept = os.path.join(self.prefix, 'lib64', 'libpil4dfs.so')
+        ior_oclass = self.params.get("ior_oclass", '/run/largefilecount/object_class/*')
+        mdtest_oclass = self.params.get("mdtest_oclass", '/run/largefilecount/object_class/*')
+
         # create pool
         self.add_pool(connect=False)
 
-        for oclass in object_class:
-            self.ior_cmd.dfs_oclass.update(oclass)
-            self.mdtest_cmd.dfs_oclass.update(oclass)
-            self.ior_cmd.dfs_dir_oclass.update(oclass)
-            # oclass_dir can not be EC must be RP based on rd_fac
-            rd_fac = extract_redundancy_factor(oclass)
-            if rd_fac >= 2:
-                self.mdtest_cmd.dfs_dir_oclass.update("RP_3G1")
-            elif rd_fac == 1:
-                self.mdtest_cmd.dfs_dir_oclass.update("RP_2G1")
-            else:
-                self.mdtest_cmd.dfs_dir_oclass.update("SX")
-            for api in apis:
-                self.ior_cmd.api.update(api)
-                self.mdtest_cmd.api.update(api)
-                # update test_dir for mdtest if api is DFS
-                if api == "DFS":
-                    self.mdtest_cmd.test_dir.update("/")
-                # run mdtest
-                if self.mdtest_cmd.api.value in ['DFS', 'POSIX']:
-                    self.log.info("=======>>>Starting MDTEST with %s and %s", api, oclass)
-                    self.container = self.add_containers(oclass)
+        for api in apis:
+            self.ior_cmd.api.update(api)
+            self.mdtest_cmd.api.update(api)
+            if api == "DFS":
+                self.mdtest_cmd.test_dir.update("/")
+            if self.mdtest_cmd.api.value in ['DFS', 'POSIX']:
+                for oclass in mdtest_oclass:
+                    self.log.info("=======>>>Starting MDTEST with %s and %s", api,
+                    oclass)
+                    self.mdtest_cmd.dfs_oclass.update(oclass)
+                    rd_fac = extract_redundancy_factor(oclass)
+                    dir_oclass = self.get_diroclass(rd_fac)
+                    self.mdtest_cmd.dfs_dir_oclass.update(dir_oclass)
+                    self.container = self.add_containers(oclass, dir_oclass)
                     try:
                         self.processes = mdtest_np
                         self.ppn = mdtest_ppn
-                        self.execute_mdtest()
+                        if self.mdtest_cmd.api.value == 'POSIX':
+                            self.mdtest_cmd.env.update(LD_PRELOAD=intercept, D_IL_REPORT='1')
```

Comment:
This is going to force all tests using this test base to use interception. Is that really what we want?

Reply:
Yeah that's what we want as of now.